apturicovid/api/StatsClient.swift

```diff
@@ -4,71 +4,93 @@ import RxSwift
 class StatsClient: RestClient {
     
     /// Specifies how long stats data is valid from .updatedAt
-    private static let statTtlInterval: TimeInterval = 2 * 24 * 60 * 60 // 2 days
+    static let statTtlInterval: TimeInterval = 2 * 24 * 60 * 60 // 2 days
     
     private static let fetchThrottleWifi: TimeInterval = 5 * 60 // 5 min
     private static let fetchThrottleCellular: TimeInterval = 10 * 60 // 10 min
     
     static let shared = StatsClient()
     private override init() { }
     
-    func getStats(forceFromApi: Bool = false, ignoreOutdated: Bool = false) -> Observable<Stats> {
+    enum FetchSource {
+        case Api, Auto, Local
+    }
+    
+    func getStats(from source: FetchSource = .Auto, ignoreOutdated: Bool = false) -> Observable<Stats> {
         
-        return Observable<Stats>.create { [weak self] observer -> Disposable in
-            
-            let disposable = Disposables.create()
-            guard let `self` = self else { return disposable }
-            
-            // Return stored data
-            if !forceFromApi {
-                if let stats = LocalStore.shared.stats {
-                    
-                    // Skip outdated if required data
-                    if ignoreOutdated {
-                        let outdated = stats.updatedAt.distance(to: Date()) > StatsClient.statTtlInterval
-                        if !outdated { observer.onNext(stats) }
-                    } else {
-                        observer.onNext(stats)
-                    }
-                }
+        let statsObservable: Observable<Stats>
+        
+        switch source {
+        case .Api:
+            statsObservable = getStatsFromApi(forced: true)
+        case .Auto:
+            statsObservable = Observable.concat([getStatsFromLocalStorage(),getStatsFromApi(forced: false)])
```

Comment:
I'd say, it's best to keep `getStatsFromApi` "dumb" and do not choose whether to actually fetch data or not - just fetch always when called.

Instead, do the logic here:

```
statsObservable = getStatsFromLocalStorage()
if (shouldFetchFromApi()) {
    statsObservable = statsObservable.concatWith(getStatsFromApi())
}
```

Reply:
Currently all stats getters use `.Auto` fetch type (except pullToRefresh). So basically .Auto **instantly** returns local available result and fetches new from API if necessary. With `.Auto` we will have multiple `.onNext` events. 

However pullToRefresh should not return local copy, as idea of pull action if to fetch new data, so it forces fetch from API, as a result up to 1 `.onNext` event

`.Local` not being used in a project, but if it will, it will return up to 1 `.onNext` event. 

If we will **not choose to fetch or not**, we will have full lot of unnecessary **duplicate fetch requests** when switching between tabs and during app delegate events. `.Auto` makes API fetches once per 5-10 min based on network connection available. 

Will change to `getStatsFromLocalStorage().concat(getStatsFromApi(forced: false))` to make it a bit more compact